testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/AbstractTestHiveViews.java

```diff
@@ -167,6 +168,45 @@ public void testSelectOnViewFromDifferentSchema()
         assertViewQuery("SELECT * FROM test_schema.hive_test_view_1", queryAssert -> queryAssert.hasRowsCount(25));
     }
 
+    @Test(groups = HIVE_VIEWS)
+    public void testViewReferencingTableInDifferentSchema()
+    {
+        String schemaX = "test_schema_x" + randomTableSuffix();
+        String schemaY = "test_schema_y" + randomTableSuffix();
+        String tableName = "test_table_y" + randomTableSuffix();
+        String viewName = "view_schema_x" + randomTableSuffix();
+
+        onHive().executeQuery(format("DROP SCHEMA IF EXISTS %s CASCADE", schemaX));
+        onHive().executeQuery(format("CREATE SCHEMA %s", schemaX));
+
+        onHive().executeQuery(format("DROP SCHEMA IF EXISTS %s CASCADE", schemaY));
+        onHive().executeQuery(format("CREATE SCHEMA %s", schemaY));
+
+        onTrino().executeQuery(format("CREATE TABLE %s.%s AS SELECT * FROM tpch.tiny.nation", schemaY, tableName));
+
+        onHive().executeQuery(format("CREATE VIEW %s AS SELECT * FROM %s.%s", viewName, schemaY, tableName));
+
+        assertThat(onTrino().executeQuery(format("SELECT COUNT(*) FROM %s", viewName))).containsOnly(row(25));
+    }
+
+    @Test(groups = HIVE_VIEWS)
+    public void testViewReferencingTableInTheSameSchemeWithoutQualifier()
+    {
+        String schemaX = "test_schema_x" + randomTableSuffix();
```

Comment:
If there's only one schema/table/view in the test you can drop the x/y parts. Maybe put the test name in the schema instead.